WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -277,22 +294,29 @@ private void ConfigureStateMachine(CoinJoinManager coinJoinManager)
 			});
 	}
 
+	private bool AutoStartTimedOut => GetRemainingTime() <= TimeSpan.Zero;
+
 	private void UpdateCountDown()
 	{
 		var format = @"hh\:mm\:ss";
 		ElapsedTime = $"{GetElapsedTime().ToString(format)}";
 		RemainingTime = $"-{GetRemainingTime().ToString(format)}";
 		ProgressValue = GetPercentage();
+
+		if (AutoStartTimedOut)
+		{
+			_stateMachine.Fire(Trigger.AutoStartTimeout);
+		}
```

Comment:
This one is checked in the `AutoStarting` state. It makes more sense there so this one can be removed IMO.

Reply:
It needs to be in both.

Reply:
You have to check when you arrive in that state if it already timed out to immediately skip through that state...
if it didnt .. you need to check it every second when you update the ui.